src/NBitcoin/IConsensus.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using NBitcoin.BouncyCastle.Math;
+
+namespace NBitcoin
+{
+    public interface IConsensus
+    {
+        /// <summary>
+        /// How many blocks should be on top of a coinbase transaction until its outputs are considered spendable.
+        /// </summary>
+        long CoinbaseMaturity { get; set; }
+
+        /// <summary>
+        /// Amount of coins mined when a new network is bootstrapped.
+        /// Set to Money.Zero when there is no premine.
+        /// </summary>
+        Money PremineReward { get; }
+
+        /// <summary>
+        /// The height of the block in which the pre-mined coins should be.
+        /// Set to 0 when there is no premine.
+        /// </summary>
+        long PremineHeight { get; }
+
+        /// <summary>
+        /// The reward that goes to the miner when a block is mined using proof-of-work.
+        /// </summary>
+        Money ProofOfWorkReward { get; }
+
+        /// <summary>
+        /// The reward that goes to the miner when a block is mined using proof-of-stake.
+        /// </summary>
+        Money ProofOfStakeReward { get; }
+
+        /// <summary>
+        /// Maximal length of reorganization that the node is willing to accept, or 0 to disable long reorganization protection.
+        /// </summary>
+        uint MaxReorgLength { get; }
+
+        /// <summary>
+        /// The maximum amount of coins in any transaction.
+        /// </summary>
+        long MaxMoney { get; }
+
+        ConsensusOptions Options { get; set; }
+
+        BuriedDeploymentsArray BuriedDeployments { get; }
+
+        BIP9DeploymentsArray BIP9Deployments { get; }
+
+        int SubsidyHalvingInterval { get; }
```

Comment:
Please comment all of those methods.

Reply:
@noescape00 all of these docs are copied straight from the existing `Consensus` object. If they are missing it's because they were missing there. I'm not sure I'm the right person to be describing what these fields are used for.

Reply:
Ok. 

It's just those values are being touched in several PRs for the last few months but they never get commented.